### PR TITLE
Use gray and italicized text for system frames

### LINKED ISSUE #50691: Stack trace: Use gray and italicized text for system frames
In order to more easily visually distinguish system vs in-app frames, use gray italicized text for the function names:

Image

### LINKED ISSUE #50693: Stack trace: Remove "System" frame tags
We want to remove this label because:

- "System" as a label doesn't always correctly describe the frame
- Since there are only two types of frames, the absence of an "In App" frame confers the same meaning
- We will have a better way to distinguish the two frame types after https://github.com/getsentry/sentry/issues/50691 is completed

Image

### PR DESCRIPTION
**Pull request by 17hogeju**
Closes https://github.com/getsentry/sentry/issues/50691 and https://github.com/getsentry/sentry/issues/50693

Distinguishes "In App" and "System" frames using gray/italicized text with no pill for "System".

Screenshot 2023-07-17 at 4 01 52 PM
Screenshot 2023-07-17 at 4 01 40 PM

### Non-Native stacktrace changes

Screenshot 2023-06-29 at 10 17 37 AM
Screenshot 2023-06-29 at 10 16 38 AM

### CODE REVIEW

**Review by scttcper — APPROVED** [review]

File: static/app/components/events/interfaces/nativeFrame.tsx [hunk 1]
```diff
@@ -346,8 +340,10 @@ const AddressCell = styled('div')`
   ${p => p.onClick && `color:` + p.theme.linkColor};
 `;
 
-const FunctionNameCell = styled('div')`
+const FunctionNameCell = styled('div')<{isSystemLabel: boolean}>`
   word-break: break-all;
+  color: ${p => (p.isSystemLabel ? p.theme.subText : 'default')};
```
Comment (malwilley) [inline static/app/components/events/interfaces/nativeFrame.tsx:345]: Is `default` a valid color/font-style? I think you want to only apply these styles conditionally, like:

```ts
${p => p.isSystemLabel
 ?  css`
      color: ${p.theme.subText};
	  font-style: 'italic';
     ` 
  : ''
```

**Comment by 17hogeju** [comment]
One thing I noticed is that when italics are used for Native Stacktrace, the location text gets cut off a little
Screenshot 2023-06-29 at 4 24 22 PM

**Comment by malwilley** [comment]
> One thing I noticed is that when italics are used for Native Stacktrace, the location text gets cut off a little
 
@17hogeju ah good catch. We do want to fix that before merging. This is happending because the containing element has `overflow: hidden` (which we use to prevent overflow and apply the ellipsis for long lib names) and it a bit too small. You'll want to find a solution where the containing element takes up the full space, or maybe even some padding might help. LMK if you need help finding the right solution.

**Review by malwilley — CHANGES_REQUESTED** [review]
Need to put this behind a feature flag, requesting changes so we don't accidentally merge

**Review by malwilley — APPROVED** [review]
Great work!

**Pull request merged**